src/main/java/com/google/sps/servlets/GenerateServlet.java

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashSet;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns bucket list content */
+@WebServlet("/generate-hunt")
+public class GenerateServlet extends HttpServlet {
+
+  private static final String FILTER_ARRAY = "clicked-array";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get array of clicked filters and convert to ArrayList<String>
+    Gson gson = new Gson();
+    HashSet<String> clickedFilters =
+        gson.fromJson(request.getParameter(FILTER_ARRAY), HashSet.class);
+
+    response.setContentType("text/html;");
+    response.getWriter().println(clickedFilters);
```

Comment:
What is the longer-term plan for this return value?  Right now you are just returning what was requested, but eventually you'll want to start the scavenger hunt if you were able to create one.  Is the plan that you get the filters, query datastore for destinations, pick some, and create a scavenger hunt, store it in datastore and... return an ID? return the scavenger hunt?  It's my understanding that @lilymzhou is working on querying the datastore for a hunt, but if her code also has the client requesting the code from the datastore, wouldn't it make more sense to just return the hunt directly from this request instead (what I suspect might be happening) only using this POST request to store the data and another GET request to get the hunt you just created from the datastore?

I want to make sure these two pieces are actually connected together and not just tangentially related via datastore.  Thanks!

Reply:
Chatted offline - @shreyachatterjee00 and @lilymzhou should meet to discuss how to make this one request flow instead of 2.